--- a/neutron/agent/linux/iptables_firewall.py
+++ b/neutron/agent/linux/iptables_firewall.py
@@ -238,6 +238,8 @@
             mac_ipv6_pairs.append((mac, ip_address))
 
     def _spoofing_rule(self, port, ipv4_rules, ipv6_rules):
+        # Disable spoofing rules for QuintupleO
+        return
         #Note(nati) allow dhcp or RA packet
         ipv4_rules += ['-p udp -m udp --sport 68 --dport 67 -j RETURN']
         ipv6_rules += ['-p icmpv6 -j RETURN']
@@ -265,6 +267,8 @@
                                        mac_ipv6_pairs, ipv6_rules)
 
     def _drop_dhcp_rule(self, ipv4_rules, ipv6_rules):
+        # Disable DHCP rules for QuintupleO
+        return
         #Note(nati) Drop dhcp packet from VM
         ipv4_rules += ['-p udp -m udp --sport 67 --dport 68 -j DROP']
         ipv6_rules += ['-p udp -m udp --sport 547 --dport 546 -j DROP']
@@ -459,8 +463,8 @@
         return iptables_rules
 
     def _drop_invalid_packets(self, iptables_rules):
-        # Always drop invalid packets
-        iptables_rules += ['-m state --state ' 'INVALID -j DROP']
+        # Always accept invalid packets for Quintupleo
+        iptables_rules += ['-m state --state ' 'INVALID -j ACCEPT']
         return iptables_rules
 
     def _allow_established(self, iptables_rules):
