diff --git a/nova/virt/libvirt/driver.py b/nova/virt/libvirt/driver.py
index 6097fbf..e419df6 100644
--- a/nova/virt/libvirt/driver.py
+++ b/nova/virt/libvirt/driver.py
@@ -4012,6 +4012,10 @@ class LibvirtDriver(driver.ComputeDriver):
             self._conf_non_lxc_uml(virt_type, guest, root_device_name, rescue,
                     instance, inst_path, image_meta, disk_info)
 
+        if (CONF.libvirt.virt_type in ['qemu', 'kvm'] and
+                flavor.extra_specs.get('libvirt:pxe-first')):
+            guest.os_boot_dev = ['network'] + guest.os_boot_dev
+
         self._set_features(guest, instance.os_type, caps, virt_type)
         self._set_clock(guest, instance.os_type, image_meta, virt_type)